core/modules/comment/comment.install

```diff
@@ -451,6 +451,17 @@ function comment_update_1006() {
   db_create_table('cache_entity_comment', $table);
 }
 
+/**
+ * Give access to the new 'Administer comment settings' permission as needed.
+ */
+function comment_update_1007() {
+  $roles = user_roles(FALSE, 'administer comments', TRUE);
+  foreach ($roles as $role) {
+    $role->permissions[] = 'administer comment settings';
+    user_role_save($role);
+  }
```

Comment:
Because our API functions can change between versions, we try to use exclusively database queries and config file modifications. See `field_update_1002()`, `file_update_1007()`, or `user_update_1013()`.

Copy/pasting one of the other functions, I think this is what is needed:

```suggestion
  $names = config_get_names_with_prefix('user.role.');
  foreach ($names as $config_name) {
    $config = config($config_name);
    $permissions = $config->get('permissions');
    if (in_array('administer comments', $permissions)) {
      $permissions[] = 'administer comment settings';
      $config->set('permissions', $permissions);
      $config->save();
    }
  }
```